_posts/articles/2021-02-14-setting-up-techdocs-on-backstage.md

```diff
@@ -0,0 +1,210 @@
+
+Transforms are then applied which modiy the generated static HTML files for a number of reasons like removing certain headers etc
+
+For the following instructions we will use the local file storage but it is better to use cloud storage like S3.
+
+## **Install Backstage**
+
+### Perquisites
+Mac running MacOS
+
+Node version 14.x
+
+cookiecutter
+
+If you already have backstage installed then skip this.
+
+```jsx
+npx @backstage/create-app
+```
+
+You are asked some questions on setup, The recommendation for this tutorial is to go with SQLLite.
+
+![npx install](/images/techdocs/Untitled.png)
+
+If successful you will see this message
+
+![successful](/images/techdocs/01.png)
+
+cd into your directory  and run 
+
+```jsx
+yarn workspace backend start
+```
+
+![yarn workspace](/images/techdocs/02.png)
+
+Open a new terminal window and 
+
+```jsx
+run yarn start
+```
+
+![yarn start](/images/techdocs/03.png)
+
+If successful a browser window will open up and you should be presented with a window
+
+![successfull register](/images/techdocs/04.png)
+
+Well done, you have successfully installed backstage on your machine.
+
+## Setup TechDocs
+
+Historically you had to manually add Techdocs , the  latest version of create-app bundles tech docs.
+
+to verify this you should be able to see entries for the following plugin 
+
+```jsx
+'@backstage/plugin-techdocs';
+  
+```
+
+in the following files
+
+```jsx
+packages/app/src/plugins.ts
+packages/app/src/App.tsx
+```
+
+## Creating and publishing techdocs
+
+To create docs manually from scratch click on create component
+
+![manuall](/images/techdocs/05.png)
+
+From here choose the Documentation template
+
+![template](/images/techdocs/06.png)
+
+Fill out Name and Description
+
+![Name Desc](/images/techdocs/07.png)
+
+Type in the owner, the Github repo you want to call it , ensure their is no Github repo that exists with the same name or the Templater will fail
+
+![owner](/images/techdocs/08.png)
+
+Once you click on create you will be presented with a Create component status popup.
+
+![create component](/images/techdocs/09.png)
+
+Once the repository has been published to github the create component status popup will go green like below.
+
+![popup](/images/techdocs/10.png)
+
+You will be able to navigate to the docs.
+
+![navigate](/images/techdocs/11.png)
+
+If it is the first time you are loading them you could receive this message while it converts from MD to html
+
+![loading](/images/techdocs/12.png)
+
+Here is what you will be presented with.
+
+![results](/images/techdocs/13.png)
+
+You now have techdocs up and running on your machine, if you want to view the files manually they are located at the following location on you machine
+
+```jsx
+backstage/node_modules/@backstage/plugin-techdocs-backend/static/docs/default/Component/
+```
+
+The recommended setup is to place the output on to cloud storage and not on the local machine
+
+## Publish to cloud storage
```

Comment:
I think we either need to omit this section and cover cloud storage elsewhere, or explain to the reader that they need to create a storage bucket.